### DISCUSSION TITLE
Problems with MAME's screen settings being shared with all shaders

### DISCUSSION THREAD

**Discussion by ICEknigh7** [mamedev/discussions#60] [Ideas]
Currently all shaders depend on the screen settings inside mame.ini for things like Brightness, Gamma, etc; settings which can't be controlled from the json files inside the chains folder. This is a big problem since those settings usually need case-by-case adjustments, for example:

Having brightness set at 1.1 is fine for simulating CRT monitors. It lets you see the black areas as not entirely black just like on real hardware and as a bonus also lets you see the limits of the screen area (useful in early games with a black background).
Since shaders like crt-geom-deluxe might not have have their own (duplicate) Brightness setting, this needs to be achieved via MAME's own option. But making this adjustment will then also greatly alter the looks of other shaders already finely-tuned to resemble things like a Game Boy Advance LCD or an Amstrad CPC green phosphor monitor.

Perhaps this problem could be overcome by having shaders handle all of these screen-related settings? Or maybe just allowing them to override MAME's values if they're also present in the json file?

**Comment by cuavas** [comment]
If you adjust brightness/contrast/gamma, isn’t it saved to the CFG file for the system independently for each emulated screen?  You can also create INI files for systems or source files with different initial settings.

**Comment by ICEknigh7** [comment]
If someone creates a shader and tweaks it till it looks spot-on, shouldn't any other MAME users around the world be able to just use that file to reproduce the exact same effect, without having to tweak any more settings for each system by themselves?

**Reply by rb6502** [reply]
That's not going to be true anyway because MAME users all around the world have different monitors in differently lit rooms.  That's exactly why it's useful to keep brightness/contrast/gamma separate from the shaders: so that the same shader settings can be made to work on other people's monitors.

**Reply by ICEknigh7** [reply]
...Aren't MAME's screen settings meant to represent the emulated display properties (complementing the shader's settings), rather than per-user emulator adjustments for the shaders to look consistent between systems?

If it's the latter, both kinds of settings (MAME's and shaders') should indeed coexist even if they're duplicated.

**Reply by ICEknigh7** [reply]
What I mean is, if you currently raise the Brightness level so that CRT shaders won't have a completely black background, any systems which don't use that shader (such as any games using an LCD display) will look too bright as a result. This could be avoided if shaders had their own settings for Brightness, Contrast, Gamma, etc.

**Reply by cuavas** [reply]
MAME saves per system/screen screen adjustments in CFG files already.  You can also set initial per-system values in per-system INI files.  Your entire complaint seems to be that global settings are global.  The only response I can think of is “well, yes, global setting are, in fact, global”.

**Reply by ICEknigh7** [reply]
You keep not getting it. Systems that use CRT shaders need one Brightness setting, other systems which need other shaders need a different setting.

Is it possible to set initial per-shader values? Because that's what's needed here.

**Comment by cuavas** [comment]
I don’t see what you’re trying to say here.  Are you talking about someone creating a tweaked shader, or someone tweaking the sliders for a shader?  If you tweak the sliders for a shader, you’d presumably share it in the form of a CFG file (where the shader slider settings are saved), which would also contain the settings for the screen container adjustment sliders.

**Comment by ICEknigh7** [comment]
I'm talking about having files inside the `chains` folder called `crt-geom-deluxe-arcade.json`, `crt-geom-deluxe-tv.json`, `crt-geom-deluxe-cpc-green.json`, `lcd-grid-gbc.json`, etc, which can be selected from the Screen Effect slider or set as default in mame.ini.

![image](https://github.com/mamedev/discussions/assets/47050710/5df49ce2-f1b6-4b02-aa26-bf015dcb2081)

For example, if you adjust MAME's Brightness to 1.1 for `crt-geom-deluxe`, it will also affect a properly adjusted `lcd-grid-gbc` and make it look too bright. The desirable behavior would be to be able to switch from one to the other and always have everything look as intended.
